src/blockMonitor/blockProcessor.ts

```diff
@@ -23,21 +44,27 @@ export class BlockProcessor extends StartStopService {
      */
     public static readonly NEW_HEAD_EVENT = "new_head";
 
-    constructor(private provider: ethers.providers.BaseProvider, private blockCache: BlockCache) {
+    constructor(private provider: ethers.providers.BaseProvider, blockCache: BlockCache) {
         super("block-processor");
 
-        this.handleBlockEvent = this.handleBlockEvent.bind(this);
+        this.mBlockCache = blockCache;
+
+        this.processBlockNumber = this.processBlockNumber.bind(this);
     }
 
     protected async startInternal(): Promise<void> {
-        this.provider.on("block", this.handleBlockEvent);
+        // Make sure the current head block is processed
+        const initialBlockNumber = await this.provider.getBlockNumber();
+        await this.processBlockNumber(initialBlockNumber);
```

Comment:
This currently emits an event on startup. Should we do this? Can we be sure that subscribers have all subscribed by the time this is called? And presumably other services will be starting up at around the same time, so the system may not full be intitialised. Maybe we should set `initialBlockNumber = await this.provider.getBlockNumber() - 1`, and not emit an event

Reply:
We should probably be actually storing the block numbers so we know where to pick up from - but I guess that's another ticket

Reply:
Good catch, I thought that the event would not be emitted. I added an additional guard to only emit the event if the service is already started.